### DISCUSSION TITLE
Drag Image Feature

### DISCUSSION THREAD

**Discussion by hjzhannah** [zauberzeug/nicegui#1339] [Q&A]
### Question

Hi! I am developing a genome browser and was wondering if there was a way to be able to implement a feature that allows dragging around within an image.

I've been playing around with ui.interactive_image() and trello cards, but I'm not quite sure how to be able to move around within the image via dragging.

**Comment by rodja** [comment]
In the [trello cards example](https://github.com/zauberzeug/nicegui/blob/main/examples/trello_cards/main.py) the card gets a ToDo model. You could either change that to be a image representation with a source string and then add an image internally to the card (`with self: ...`) or you remove the model passing code and simply use the `dnd.card` as a normal card:

```py
with dnd.card().tight():
    ui.image('/some/image.png`)
```

You should also be able to change the inheritance from `dnd.card()` from `ui.card` to `ui.image` to get a draggable image without any "container".

**Comment by rodja** [comment]
I think I misunderstood you. My answer was about dragging images around. But you asked to implement dragging inside an image. That should be possible with `ui.interactive_image`. You need to implement the proper mouse event handlers and draw/update the things you want to drag with svg inside the `.content` property.

**Reply by hjzhannah** [reply]
I will give this a try, thank you so much!
